src/pip/_internal/commands/list.py

```diff
@@ -303,8 +303,9 @@ def format_for_json(packages, options):
             'name': dist.project_name,
             'version': six.text_type(dist.version),
         }
-        if options.verbose >= 1:
+        if options.verbose >= 1 or dist_is_editable(dist):
             info['location'] = dist.location
+        if options.verbose >= 1:
             info['installer'] = get_installer(dist)
```

Comment:
It does not feel right to me to have the verbose flag influence the content of a json output.
What is also not completely right is to have the location field having two semantics: the editable project location and the installed location. These are two different things that should go in different fields IMO.

Reply:
The location semantic would also affect the column format. It has also existed for a while now so the compatibility issue would be important.

Regarding verbosity, I have no problem always outputting all fields.

Reply:
I *think* the concern was backwards compatibility?

If this mode of output does not feel very reasonable, I don't mind changing it but I'm also not sure if there's a clear transition path to introduce these additional fields and who it might affect if we do so.

Reply:
Yeah, agreed. One solution I can think of is to introduce a `--fields` option that explicitly tells pip what fields to output (the default being to infer from verbosity).

Reply:
I think our tests breaking because of new fields does not necessarily indicate that there are people out there relying on the absence of such fields. So I'd still be inclined to add the fields unconditionally and adapting our tests.